src/app/hooks/useClickTracker/index.test.jsx

```diff
@@ -0,0 +1,322 @@
+import pidginData from './fixtureData/tori-51745682.json';
+import zhongwenData from './fixtureData/chinese-news-49631219-trad.json';
+import { RequestContextProvider } from '#contexts/RequestContext';
+import { ServiceContextProvider } from '#contexts/ServiceContext';
+import { STORY_PAGE } from '#app/routes/utils/pageTypes';
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+jest.useFakeTimers();
+
+const urlToObject = url => {
+  const { origin, pathname, searchParams } = new URL(url);
+
+  return {
+    origin,
+    pathname,
+    searchParams: Object.fromEntries(searchParams),
+  };
+};
+
+delete window.location;
+window.location = { href: 'http://bbc.com/pidgin/tori-51745682' };
+process.env.SIMORGH_ATI_BASE_URL = 'https://logws1363.ati-host.net?';
+
+const defaultProps = {
+  pageData: pidginData,
+  componentName: 'brand',
+};
+
+const WithContexts = ({
+  children,
+  variant,
+  service = 'pidgin',
+  pathname = '/pidgin/tori-51745682',
+}) => (
+  <RequestContextProvider
+    bbcOrigin="https://www.test.bbc.com"
+    pageType={STORY_PAGE}
+    isAmp={false}
+    pathname={pathname}
+    service={service}
+    variant={variant}
+  >
+    <ServiceContextProvider service={service}>
+      {children}
+    </ServiceContextProvider>
+  </RequestContextProvider>
+);
+
+const TestComponentContainerContainer = () => {
+  const clickRef = useClickTracker({
+    pageData: pidginData,
+    componentName: 'header',
+  });
+
+  return (
+    <div ref={clickRef}>
+      <TestComponentContainer hookProps={defaultProps} />
+    </div>
+  );
+};
+
+const TestComponentContainer = ({ hookProps }) => {
+  const clickRef = useClickTracker(hookProps);
+  return (
+    <div>
+      <TestComponent ref={clickRef} />
+    </div>
+  );
+};
+
+const TestComponent = React.forwardRef((_, ref) => {
+  return (
+    <div id="test-component" ref={ref}>
+      <a href="https://www.test.bbc.com">Testing Link</a>
+      <button type="button" id="test-button">
+        Testing Button
+      </button>
+    </div>
+  );
+});
+
+// eslint-disable-next-line react/prop-types
+const wrapper = ({ children }) => <WithContexts>{children}</WithContexts>;
+
+describe('Click tracking', () => {
+  it('should return a ref', () => {
+    const { result } = renderHook(() => useClickTracker(defaultProps), {
+      wrapper,
+    });
+
+    const { container } = render(
+      <WithContexts>
+        <TestComponent ref={result.current} />
+      </WithContexts>,
+    );
+
+    expect(result.current.current).toBe(
+      container.querySelector('#test-component'),
+    );
+  });
+
+  it('should send a single tracking request on click', async () => {
+    const spyFetch = jest.spyOn(global, 'fetch');
+    const { container } = render(
+      <WithContexts>
+        <TestComponentContainer hookProps={defaultProps} />
+      </WithContexts>,
+    );
+
+    expect(spyFetch).toHaveBeenCalledTimes(0);
+
+    act(() => container.querySelector('#test-component').click());
```

Comment:
I think it might be better to use the query and event utilities provided react-tesing-library e.g.

```jsx
const { getByText } = render(
  <WithContexts>
    <TestComponentContainer hookProps={defaultProps} />
  </WithContexts>,
);

const linkEl = getByText('Test Link')
const buttonEl = getByText('Test  Button')
```

```js
import userEvent from '@testing-library/user-event'

act(() => userEvent.click(linkEl));

act(() => userEvent.click(buttonEl));
```
More on react-testing-library queries and events:
- https://testing-library.com/docs/queries/about/
- https://testing-library.com/docs/ecosystem-user-event/#clickelement-eventinit-options

Reply:
Makes sense, will do this now